virtcontainers/network.go

```diff
@@ -1277,6 +1543,47 @@ func generateInterfacesAndRoutes(networkNS NetworkNamespace) ([]*grpc.Interface,
 	return ifaces, routes, nil
 }
 
+func createNetworkInterfacePair(idx int, ifName string, interworkingModel NetInterworkingModel) (NetworkInterfacePair, error) {
+	uniqueID := uuid.Generate().String()
+
+	randomMacAddr, err := generateRandomPrivateMacAddr()
+	if err != nil {
+		return NetworkInterfacePair{}, fmt.Errorf("Could not generate random mac address: %s", err)
+	}
+
+	netPair := NetworkInterfacePair{
+		ID:   uniqueID,
+		Name: fmt.Sprintf("br%d_kata", idx),
+		VirtIface: NetworkInterface{
+			Name: fmt.Sprintf("eth%d", idx),
+			//HardAddr: hardAddr.String(),
+			HardAddr: randomMacAddr,
+		},
+		TAPIface: NetworkInterface{
+			Name: fmt.Sprintf("tap%d_kata", idx),
+		},
+		NetInterworkingModel: interworkingModel,
+	}
+
+	return netPair, nil
+}
+
+func generateRandomPrivateMacAddr() (string, error) {
```

Comment:
This could do with a unit test, particularly since it's cryptographically sensitive.

Reply:
Yup. Will add more unit tests overall.

Reply:
fixed.